### PR TITLE
Update to use remark-mermaidjs

### LINKED ISSUE #5: Allow configuration of puppeteer launch options
I run gatsby build in alpine-based Docker and puppeteer has some limitations there: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-on-alpine

Here is how I start puppeteer: 

```
const browser = await puppeteer.launch({
    executablePath: process.env.CHROMIUM_PATH,
    args: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-dev-shm-usage'],
  });
```

It's ok if you close it as out of scope for this project.

### LINKED ISSUE #15: How I got this plugin to work with Puppeteer 5
I'm using Puppeteer 5.2.1 for testing and with that dependency, gatsby-remark-mermaid generated incomplete diagrams. I changed the following settings and it worked with Puppeteer 5. I'd send a PR but I'm not sure if these are correct for everyone.

```
browser = await puppeteer.launch({args: [
  '--disable-gpu',
  '--disable-dev-shm-usage',
  '--disable-setuid-sandbox',
  '--no-first-run',
  '--no-sandbox',
  '--no-zygote',
  '--single-process',
]});
```

### LINKED ISSUE #17: Adding the configurations for puppeteer
The mermaid diagrams are rendered properly on my end, however it gets an error on the build. Here is the error message:
```
Failed to launch the browser process!
/tmp/build/node_modules/puppeteer/.local-chromium/linux-970485/chrome-linux/chrome: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
```
I looked into the documentation and since I'm running it on Heroku, I must include this configuration:
```
puppeteer.launch({ args: ['--no-sandbox'] });
```
I tried using it on `gatsby-browser.js` like this, but I only got errors instead.
```javascript
const puppeteer = require('puppeteer')
const browser = await puppeteer.launch({ args: ['--no-sandbox'] })
```
I'm not really familiar with `puppeteer`, so where do I need to put this configuration for it to work?

### PR DESCRIPTION
**Pull request by bdenham**
## Description

This PR provides the following updates:

- Updates `index.js` to import and use `remark-mermaidjs`. 
- Updates `package.json` to add `remark-mermaidjs` as the only dependency.
- Updates `README` docs to describe all the updates.

## How to test this PR

1. `git clone https://github.com/bdenham/mdx-starter.git`
2. `cd mdx-starter`
3. `git checkout `remark-mermaidjs-test`
4. `yarn install`
5. `yarn dev`
6. Click "[Mermaid MDX test page.](http://localhost:8000/mermaid/)" in the browser window.
7. You should see all the same mermaid diagrams currently supported on [https://mermaid.live/](https://mermaid.live/) rendered on this page using the mermaid `base` theme: Flowchart, Sequence, Gantt, Class, State, Pie Chart, Git Graph, User Journey, and Entity Relationship. 
8. Compare the test page rendering of these diagrams to the [https://mermaid.live/](https://mermaid.live/) online editor with the editor's config set to `{"theme": "base"}`. 
9. Open the test project in your IDE and open the `gatsby-config.js` file.
10. Change the [mermaid theme config setting](https://github.com/bdenham/mdx-starter/blob/861237aeb8d177520a377e1afb3fb466237e7436/gatsby-config.js#L48) to one of the other built-in mermaid themes, named and [described here](https://mermaid-js.github.io/mermaid/#/theming?id=deployable-themes).
11. Open the test project in your IDE and change or add your own mermaid diagrams to the test file: `src/pages/mermaid.mdx`.

### Other testing details

Updates to the package were manually tested in both Gatsby v4 and v5 projects:

- Gatsby v5 MDX starter project
- Gatsby v4.24  theme project 

In both projects, I used [yalc](https://www.npmjs.com/package/yalc) to test my local package changes to ensure I could successfully render the same markdown file (see below) containing all the currently supported mermaid diagrams.

### Gatsby v5 MDX starter project

**See instructions above to run this test.**

The test markdown file was added to the Gatsby v5 starter project using both `.md` and `.mdx` extensions. All diagrams rendered correctly, matching the output shown on [https://mermaid.live/](https://mermaid.live/). All plugin options worked as designed using the following configuration:

```js
{
  resolve: 'gatsby-remark-mermaid',
  options: {
    language: 'mermaid',
    theme: 'base',
    viewport: {
      width: 1280,
      height: 800
    },
    mermaidOptions: {
      themeVariables: mermaidThemeVariables,
      themeCSS: mermaidThemeCSS,
      flowchart: flowchart,
    },
  },
},
```

### Gatsby v4.24  theme project

The test markdown file was added to the Gatsby v4.24 example project within the theme's monorepo. All diagrams rendered correctly, matching the output shown on [https://mermaid.live/](https://mermaid.live/). All plugin options worked as designed using the same configuration noted above.

## Markdown Test File

<details><summary>MARKDOWN TEST FILE — Edit this description to copy it.</summary>

# Mermaid Tests

This is a test page for mermaid diagrams.

## Flowchart

```mermaid
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
```

### From Live Editor

  ```mermaid
  sequenceDiagram
      Alice->>+John: Hello John, how are you?
      Alice->>+John: John, can you hear me?
      John-->>-Alice: Hi Alice, I can hear you!
      John-->>-Alice: I feel great!
  ```

### With loop

```mermaid
sequenceDiagram
    Alice->>John: Hello John, how are you?
    loop Healthcheck
        John->>John: Fight against hypochondria
    end
    Note right of John: Rational thoughts!
    John-->>Alice: Great!
    John->>Bob: How about you?
    Bob-->>John: Jolly good!
```

## Gantt Diagram

```mermaid
gantt
    dateFormat  YYYY-MM-DD
    title Adding GANTT diagram functionality to mermaid

    section A section
    Completed task            :done,    des1, 2014-01-06,2014-01-08
    Active task               :active,  des2, 2014-01-09, 3d
    Future task               :         des3, after des2, 5d
    Future task2              :         des4, after des3, 5d

    section Critical tasks
    Completed task in the critical line :crit, done, 2014-01-06,24h
    Implement parser and jison          :crit, done, after des1, 2d
    Create tests for parser             :crit, active, 3d
    Future task in critical line        :crit, 5d
    Create tests for renderer           :2d
    Add to mermaid                      :1d

    section Documentation
    Describe gantt syntax               :active, a1, after des1, 3d
    Add gantt diagram to demo page      :after a1  , 20h
    Add another diagram to demo page    :doc1, after a1  , 48h

    section Last section
    Describe gantt syntax               :after doc1, 3d
    Add gantt diagram to demo page      : 20h
    Add another diagram to demo page    : 48h
```

## Class Diagram

```mermaid
classDiagram
    Animal <|-- Duck
    Animal <|-- Fish
    Animal <|-- Zebra
    Animal : +int age

    Animal : +String getSpecies()
    Animal : +void eat()
    class Duck{
        +String beakColor
        +swim()
        +quack()
    }
    class Fish{
        -int sizeInFeet
        -canEat()
    }
    class Zebra{
        +bool is_wild
        +run()
    }
```

## State Diagram

```mermaid
stateDiagram
    [*] --> Still
    Still --> [*]

    Still --> Moving
    Moving --> Still
    Moving --> Crash
    Crash --> [*]
```

## Pie Chart

```mermaid

pie
    title Key elements in Product X
    "Calcium" : 42.96
    "Potassium" : 50.05
    "Magnesium" : 10.01
    "Iron" :  5
```

## Git Graph

```mermaid
gitGraph
    commit
    commit
    branch develop
    checkout develop
    commit
    commit
    checkout main
    merge develop
    commit
    commit
```

## User Journey

```mermaid
journey
    title My working day
    section Go to work
      Make tea: 5: Me
      Go upstairs: 3: Me
      Do work: 1: Me, Cat
    section Go home
      Go downstairs: 5: Me
      Sit down: 5: Me
```

## Entity Relationship Diagram

```mermaid
erDiagram
    CUSTOMER }|..|{ DELIVERY-ADDRESS : has
    CUSTOMER ||--o{ ORDER : places
    CUSTOMER ||--|{ INVOICE : "liable for"
    DELIVERY-ADDRESS ||--o{ ORDER : receives
    INVOICE ||--|{ ORDER : covers
    ORDER ||--|{ ORDER-ITEM : includes
    PRODUCT-CATEGORY ||--|{ PRODUCT : contains
    PRODUCT ||--o{ ORDER-ITEM : "ordered in"
```
</details>

Closes #5
Closes #6
Closes #15
Closes #17

### CODE REVIEW

**Comment by remcohaszing** [comment]
I suggest to revamp this plugin entirely using [`remark-mermaidjs`](https://github.com/remcohaszing/remark-mermaidjs) instead.

I think all that’s needed is the following code:

```js
async function gatsbyRemarkMermaid(ast, options) {
  const { default: plugin } = await import('remark-mermaidjs')
  const transformer = plugin(options)
  return transformer(ast)
}

module.exports = gatsbyRemarkMermaid
```

Could you give that a try?

Technically the transformer takes a second argument `vfile`, but I’m not sure how to retrieve it from the gatsby interface and it’s unused by `remark-mermaidjs`.

**Comment by bdenham** [comment]
@remcohaszing: Take a look. Using `remark-mermaidjs` keeps this plugin pretty simple. I do have a suggestion to for updating `remark-mermaidjs` so we could pass in some custom viewport settings that would look something like this when configuring this plugin on the Gatsby side of things:

```
{
  resolve: `gatsby-remark-mermaid`,
  options: {
    launchOptions: {
      executablePath: puppeteer.executablePath(),
      defaultViewport: {
        width: 1280,
        height: 3000,
      }
    },
  }
}
```

File: README.md [hunk 1]
```diff
@@ -1,90 +1,89 @@
 # gatsby-remark-mermaid
 [![npm](https://img.shields.io/npm/v/gatsby-remark-mermaid.svg?style=flat-square)](https://www.npmjs.com/package/gatsby-remark-mermaid)
 
-Create [mermaid](https://mermaidjs.github.io/) graphs and diagrams in your markdown files.
+Creates [mermaid](https://mermaidjs.github.io/) graphs and diagrams in your markdown files.
```
Comment (remcohaszing) [inline README.md]: I prefer an imperative style tag line.

```suggestion
Create [mermaid](https://mermaidjs.github.io/) graphs and diagrams in your markdown files.
```

File: README.md [hunk 2]
```diff
@@ -1,90 +1,89 @@
...
 
-This plugin uses **server-side rendering**. This means the svg is rendered on build time instead of having a runtime
-dependency on mermaid.
+This plugin uses [remark-mermaidjs](https://github.com/remcohaszing/remark-mermaidjs) to generate SVG diagrams at build-time. The mermaid code blocks are replaced with an inline SVGs in the generated HTML. This prevents any runtime
```
Comment (remcohaszing) [inline README.md]:
```suggestion
This plugin uses [remark-mermaidjs](https://github.com/remcohaszing/remark-mermaidjs) to generate SVG diagrams at build time. The mermaid code blocks are replaced with an inline SVG in the generated HTML. This prevents any runtime
```

File: README.md [hunk 3]
```diff
@@ -1,90 +1,89 @@
...
+dependencies on `mermaid.js`.
 
 ## Install
 
-`npm install --save gatsby-remark-mermaid gatsby-transformer-remark puppeteer`
+```bash
```
Comment (remcohaszing) [inline README.md]:
```suggestion
```sh
```

File: README.md [hunk 4]
```diff
@@ -1,90 +1,89 @@
...
+npm install gatsby-remark-mermaid gatsby-transformer-remark
+```
+
+## How to use
```
Comment (remcohaszing) [inline README.md]: Thank you for structuring the readme!

I’ve been trying to use a more consistent readme layout across my repositories, including consistent headers.

```suggestion
## Usage
```

File: README.md [hunk 5]
```diff
@@ -1,90 +1,89 @@
...
-    {
-      resolve: 'gatsby-transformer-remark',
-      options: {
-        plugins: [
-          'gatsby-remark-mermaid'
-        ]
-      }
+// In your gatsby-config.js
+
+plugins: [
```
Comment (remcohaszing) [inline README.md]: Let’s provide fully valid JavaScript, including:

```js
module.exports = {
  // …
}
```

File: README.md [hunk 6]
```diff
@@ -1,90 +1,89 @@
...
+      ]
     }
-  ]
-}
+  }
+],
 ```
 
-Now you can use markdown:
+## Configuration Options
```
Comment (remcohaszing) [inline README.md]:
```suggestion
## Options
```

File: README.md [hunk 7]
```diff
@@ -1,90 +1,89 @@
...
+However, the table below describes the configuration details as they apply to Gatsby.
 
-To generate:
+| Name                            | Default              | Description                                                                                                                                                                                                                                                               |
+| :------------------------------ | :------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `launchOptions.executablePath`  | `""`                 | (Required) String path to the chrome executable that puppeteer uses to render the mermaid diagrams to SVGs.                                                                                                                                                               |
```
Comment (remcohaszing) [inline README.md]: This option has no default value.

```suggestion
| `launchOptions.executablePath`  |                      | (Required) String path to the chrome executable that puppeteer uses to render the mermaid diagrams to SVGs.                                                                                                                                                               |
```

File: README.md [hunk 8]
```diff
@@ -1,90 +1,89 @@
...
+| `mermaidOptions.themeCSS`       | `""`                 | (Optional) Override mermaid styles using `themeCSS`. See [mermaid themes](https://mermaid-js.github.io/mermaid/#/Setup?id=theme).                                                                                                                                         |
+| `mermaidOptions.themeVariables` | `{}`                 | (Optional) Override mermaid variables using `themeVariables`. See [Customizing Themes with themeVariables](https://mermaid-js.github.io/mermaid/#/./theming?id=customizing-themes-with-themevariablesSpecifies).)                                                         |
+| `svgo.plugins`                  | `defaultSVGOOptions` | (Optional) Override default optimizations for the generated SVG files. Set to `null` to disable minifying using SVGO completely. See [defaultSVGOOptions](https://github.com/remcohaszing/remark-mermaidjs/blob/529e44babf07b593489efbe7c238a39089066faa/index.ts#L18)).) |
```
Comment (remcohaszing) [inline README.md]:
```suggestion
| `svgo.plugins`                  | `defaultSVGOOptions` | (Optional) Override default optimizations for the generated SVG files. Set to `null` to disable minifying using SVGO completely. See [defaultSVGOOptions](https://github.com/remcohaszing/remark-mermaidjs/blob/v4.0.0/index.ts#L18)).) |
```

File: package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gatsby-remark-mermaid",
-  "version": "2.1.0",
+  "version": "2.2.0",
```
Comment (remcohaszing) [inline package.json]: Could you revert this version? This is part of the release process. It should not be part of a PR.

```suggestion
  "version": "2.1.0",
```

File: package.json [hunk 10]
```diff
@@ -25,10 +25,6 @@
...
   "dependencies": {
-    "mermaid": "^8.7.0",
-    "unist-util-visit": "^1.4.0"
-  },
-  "peerDependencies": {
-    "puppeteer": "*"
+    "remark-mermaidjs": "^3.1.0"
   }
-}
+}
```
Comment (remcohaszing) [inline package.json]:
```suggestion
}

```

File: package.json [hunk 11]
```diff
@@ -25,10 +25,6 @@
   },
   "homepage": "https://github.com/remcohaszing/gatsby-remark-mermaid#readme",
   "dependencies": {
-    "mermaid": "^8.7.0",
-    "unist-util-visit": "^1.4.0"
-  },
-  "peerDependencies": {
-    "puppeteer": "*"
+    "remark-mermaidjs": "^3.1.0"
```
Comment (remcohaszing) [inline package.json]: I just created a new release of `remark-mermaidjs`. Let’s use that here. It works pretty much the same ([release notes](https://github.com/remcohaszing/remark-mermaidjs/releases/tag/v4.0.0))

```suggestion
    "remark-mermaidjs": "^4.0.0"
```

**Review by remcohaszing — CHANGES_REQUESTED** [review]
Thank you! This looks really great!

I just have a few nit picks.

**Comment by bdenham** [comment]
@remcohaszing: Great review! It's a true pleasure to work with someone who pays attention to detail!

**Comment by remcohaszing** [comment]
Awesome! Thank you! :heart:

**Comment by bdenham** [comment]
Thanks, @remcohaszing! I look forward to working with you more in the future!

**Comment by remcohaszing** [comment]
Aww thanks! I’m sure we will if you stay around in the remark / MDX ecosystem. :)

**Pull request merged**